src/frontend/http.rs

```diff
@@ -108,12 +110,21 @@ async fn physical_plan_to_json(
     Ok(buf)
 }
 
-/// POST /q
+/// POST /q or /[database_name]/q
 pub async fn uncached_read_write_query(
+    database_name: Option<String>,
     user_context: UserContext,
     query: String,
-    context: Arc<dyn SeafowlContext>,
+    mut context: Arc<DefaultSeafowlContext>,
 ) -> Result<Vec<u8>, ApiError> {
+    // If a specific DB name was used as a parameter in the route, scope the context to it,
```

Comment:
We might need to create the DB lazily here on first write, also because if we need to use `/q` to create the DB first and there are a lot of databases, the context will have to load all of their tables/schemas.

Reply:
As mentioned context loads schemas/tables for a single DB only, and this is the case also for the default route `/q`, i.e. we load `DEFAULT_DB` stuff in the `reload_schema`. Also, note that `CREATE DATABASE` works in any reference system, so I don't think we need to create the DB here.

Reply:
Got it -- so for the replication use case would we be able to somehow determine if a DB exists and then create it if not, e.g. by trying to first `POST` a `SELECT 1` to `/q/db-id` or by hitting the information schema?

Reply:
Yeah, `POST` of `SELECT 1` to `/db-id/q` would work for that. Alternatively, catch `Unknown database my-db; try creating one with CREATE DATABASE first` error during initial replication attempt, create the DB and then retry replication.

Another option would be to have a separate endpoint to list all available DBs.